tests/unit_node/crypto/crypto_hkdf_test.ts

```diff
@@ -34,3 +34,271 @@ Deno.test("crypto.hkdfSync - compare with node", async (t) => {
+  assertEquals(
+    dataViewHex,
+    "f6d2fcc47cb939deafe3",
+    "DataView should produce expected Node.js result",
+  );
+
+  // Test DataView with different underlying buffer scenarios
+  const largerBuffer = new ArrayBuffer(16);
+  const view = new Uint8Array(largerBuffer);
+  view.set([115, 97, 108, 116], 4); // "salt" at offset 4
+
+  const offsetDataView = new DataView(largerBuffer, 4, 4);
+  const offsetResult = hkdfSync(
+    "sha256",
+    secret,
+    offsetDataView,
+    "info",
+    keyLen,
+  );
+
+  assertEquals(
+    Buffer.from(offsetResult).toString("hex"),
+    stringHex,
+    "DataView with offset should still produce correct result",
+  );
+});
+
+Deno.test("crypto.hkdfSync - edge cases and error conditions", () => {
+  const secret = "secret";
+  const keyLen = 10;
+
+  // Test empty TypedArrays
+  const emptyResult = hkdfSync(
+    "sha256",
+    secret,
+    new Uint8Array(0), // Empty salt
+    new Uint8Array(0), // Empty info
+    keyLen,
+  );
+  assertEquals(emptyResult.byteLength, keyLen, "Empty TypedArrays should work");
+
+  // Test single-byte TypedArrays
+  const singleByteResult = hkdfSync(
+    "sha256",
+    secret,
+    new Uint8Array([65]), // Single byte 'A'
+    new Uint8Array([66]), // Single byte 'B'
+    keyLen,
+  );
+  assertEquals(
+    singleByteResult.byteLength,
+    keyLen,
+    "Single-byte TypedArrays should work",
+  );
+
+  // Test maximum allowed info length (1024 bytes)
+  const maxInfo = new Uint8Array(1024).fill(42);
+  const maxInfoResult = hkdfSync("sha256", secret, "salt", maxInfo, keyLen);
+  assertEquals(
+    maxInfoResult.byteLength,
+    keyLen,
+    "Maximum info length should work",
+  );
+
+  // Test info length exceeding limit should throw
+  const oversizedInfo = new Uint8Array(1025).fill(42);
+  let threwError = false;
+  try {
+    hkdfSync("sha256", secret, "salt", oversizedInfo, keyLen);
+  } catch (error) {
+    threwError = true;
+    assertEquals(
+      (error as Error).message.includes(
+        "must not contain more than 1024 bytes",
+      ),
+      true,
+      "Should throw specific error for oversized info",
+    );
+  }
+  assertEquals(threwError, true, "Oversized info should throw error");
+
+  // Test various key lengths
+  const keyLengths = [1, 32, 64, 255];
+  for (const len of keyLengths) {
+    const result = hkdfSync("sha256", secret, "salt", "info", len);
+    assertEquals(result.byteLength, len, `Key length ${len} should work`);
+  }
+
+  // Test different digest algorithms with TypedArrays
+  const digests = ["sha1", "sha256", "sha384", "sha512"];
+  for (const digest of digests) {
+    const result = hkdfSync(
+      digest,
+      secret,
+      new Uint16Array(Buffer.from("salt")),
+      new Uint16Array(Buffer.from("info")),
+      keyLen,
+    );
+    assertEquals(
+      result.byteLength,
+      keyLen,
+      `${digest} with TypedArrays should work`,
+    );
+  }
+});
+
+// TODO: Consider adding async hkdf() testing in a future PR
```

Comment:
Is it possible to address these in the PR? (ie. make sure async version works correctly too)